### PR TITLE
Kp issue4

### LINKED ISSUE #9: Image Tags
#### As a user I want to add tags to images.

__Story__:

Our users are loving the ability to add images to our service. However, now
that there are more than a handful of images, it is apparent that we need some
way to filter images by certain categories. In order to do that, we first need
to label images in some way. It is time to add the ability for our users to tag
images when they add them.

__Acceptance criteria__:
- [ ] When I add a new image, there is a form field to add tags.
- [ ] After I've saved an image with tags, I see those tags displayed with the
  associated image.
- [ ] My application uses the acts-as-taggable-on gem to provide the tagging
  functionality.

__Reference material__:
- https://github.com/mbleigh/acts-as-taggable-on

__Note__: You should allow images to have no tags; this sets the stage for a later issue.

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by sthust**
Closes #9

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

**Comment by sthust** [comment]
[Issue 4: Image Tags](https://trello.com/c/uaiRw88U/718-issue-4-image-tags)

File: app/models/image.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
 class Image < ApplicationRecord
+  acts_as_taggable
```
Comment (natalieschauser) [inline app/models/image.rb]: I think we can split this out into a separate commit and add some tests as well.

File: app/views/images/index.html.erb [hunk 2]
```diff
@@ -8,7 +8,7 @@
   <% @images.each_with_index do |image, index| %>
     <tr>
       <td>
-        <img id="display_image<%= index %>" width='400' src="<%= image.url %>">
+        <img id="display_image<%= index %>" width='400' src="<%= image.url %>"><a id="tags<%= index %>"><%= image.tag_list %></a>
```
Comment (natalieschauser) [inline app/views/images/index.html.erb]: Same here, I think we can split this out into a separate commit. So maybe the commits can be
1. the model changes
2. the new/create/show changes
3. the index changes

File: app/views/images/show.html.erb [hunk 3]
```diff
@@ -6,4 +6,7 @@
   <p id="notice"> <%= flash[:notice] %> </p>
 <%end%>
 
-<img id="display_image" src="<%= @image.url %>">
+<a><%= t :image_input_form_field_tag_label %></a>  <a id="tags"><%= @image.tag_list %></a>
```
Comment (natalieschauser) [inline app/views/images/show.html.erb]: Can we add a test where the image does not have any tags?

File: app/views/images/index.html.erb (hunk 2 above)
Comment (natalieschauser) [reply app/views/images/index.html.erb]: Can we also add another test where there are images that don't have tags and make sure we don't see anything links for these images?

File: test/controllers/images_controller_test.rb [hunk 4]
```diff
@@ -7,29 +7,42 @@ def test_should_get_index
     assert_select '#title', 'Upload an image'
   end
 
+  def test_index_descending_order
```
Comment (natalieschauser) [inline test/controllers/images_controller_test.rb]: For complete atomic commits, this test should be moved into the next commit :)
